ogr/ogrsf_frmts/dwg/ogrdwglayer.cpp

```diff
@@ -1128,6 +1128,38 @@ class GeometryInsertTransformer : public OGRCoordinateTransformation
         }
 };
 
+/************************************************************************/
+/*                        Translate3DFACE()                             */
+/************************************************************************/
+
+OGRFeature* OGRDWGLayer::Translate3DFACE(OdDbEntityPtr poEntity)
+{
+    OGRFeature* poFeature = new OGRFeature(poFeatureDefn);
+    OdDbFacePtr pFace = OdDbFace::cast(poEntity);
+
+    TranslateGenericProperties(poFeature, poEntity);
+
+    /* -------------------------------------------------------------------- */
+    /*      Create a polygon geometry from the vertices.                    */
+    /* -------------------------------------------------------------------- */
+    OGRPolygon* poPolygon = new OGRPolygon();
+
+    OGRLinearRing* poLinearRing = new OGRLinearRing();
+
+    OdInt16 index;
+    OdGePoint3d point;
+
+    for (index = 0; index <= 3; index++)
+    {
+        pFace->getVertexAt(index, point);
+        poLinearRing->addPoint(point.x,point.y,point.z);
+    }
+    poPolygon->addRingDirectly(poLinearRing);
```

Comment:
I assume that point at index 3 is the same as index 0 ? Otherwise you should call closeRings() to add as last point the first point

Reply:
ok fine, i apply the closeRings check